modules/gapi/src/executor/gstreamingexecutor.hpp

```diff
@@ -138,75 +60,22 @@ class GStreamingExecutor final
     util::optional<bool> m_reshapable;
 
     cv::gimpl::GIslandModel::Graph m_gim; // FIXME: make const?
-    const bool m_desync;
-
-    // FIXME: Naive executor details are here for now
-    // but then it should be moved to another place
-    struct OpDesc
-    {
-        std::vector<RcDesc> in_objects;
-        std::vector<RcDesc> out_objects;
-        cv::GMetaArgs       out_metas;
-        ade::NodeHandle     nh;
-
-        cv::GRunArgs in_constants;
-
-        std::shared_ptr<GIslandExecutable> isl_exec;
-    };
-    std::vector<OpDesc> m_ops;
-
-    struct DataDesc
-    {
-        ade::NodeHandle slot_nh;
-        ade::NodeHandle data_nh;
-    };
-    std::vector<DataDesc> m_slots;
 
     cv::GRunArgs m_const_vals;
 
-    // Order in these vectors follows the GComputaion's protocol
-    std::vector<ade::NodeHandle> m_emitters;
-    std::vector<ade::NodeHandle> m_sinks;
-
-    class Synchronizer;
-    std::unique_ptr<Synchronizer> m_sync;
-
-    std::vector<std::thread> m_threads;
-    std::vector<stream::SyncQueue>   m_emitter_queues;
-
-    // a view over m_emitter_queues
-    std::vector<stream::SyncQueue*>  m_const_emitter_queues;
-
-    std::vector<stream::Q*>          m_sink_queues;
-
-    // desync path tags for outputs. -1 means that output
-    // doesn't belong to a desync path
-    std::vector<int>                 m_sink_sync;
-
-    std::unordered_set<stream::Q*>   m_internal_queues;
-    stream::SyncQueue m_out_queue;
-
-    // Describes mapping from desync paths to collector threads
-    struct CollectorThreadInfo {
-        std::vector<stream::Q*>  queues;
-        std::vector<int> mapping;
-    };
-    std::unordered_map<int, CollectorThreadInfo> m_collector_map;
-
-
-    void wait_shutdown();
-
-    cv::GTypesInfo out_info;
+    void virtual start_impl() = 0;
```

Comment:
Could you, please, explain why `_impl`s are `virtual` and not the default functions?